src/Raven.Studio/typescript/viewmodels/database/settings/customAnalyzers.ts

```diff
@@ -2,60 +2,72 @@ import viewModelBase = require("viewmodels/viewModelBase");
 import appUrl = require("common/appUrl");
 import getCustomAnalyzersCommand = require("commands/database/settings/getCustomAnalyzersCommand");
 import deleteCustomAnalyzerCommand = require("commands/database/settings/deleteCustomAnalyzerCommand");
+import getServerWideCustomAnalyzersCommand = require("commands/serverWide/analyzers/getServerWideCustomAnalyzersCommand");
 import database = require("models/resources/database");
 import router = require("plugins/router");
 import aceEditorBindingHandler = require("common/bindingHelpers/aceEditorBindingHandler");
 import generalUtils = require("common/generalUtils");
+import analyzerListItemModel = require("models/database/settings/analyzerListItemModel");
+import accessManager = require("common/shell/accessManager");
 
-class analyzerListItem {
-    
-    definition: Raven.Client.Documents.Indexes.Analysis.AnalyzerDefinition;
-    
-    constructor(dto: Raven.Client.Documents.Indexes.Analysis.AnalyzerDefinition) {
-        this.definition = dto;
-    }
-    
-    get name() {
-        return this.definition.Name;
-    }
-}
 
 class customAnalyzers extends viewModelBase {
-    analyzers = ko.observableArray<analyzerListItem>([]);
+    analyzers = ko.observableArray<analyzerListItemModel>([]);
+    serverWideAnalyzers = ko.observableArray<analyzerListItemModel>([]);
     
     addUrl = ko.pureComputed(() => appUrl.forEditCustomAnalyzer(this.activeDatabase()));
     
-    isFirstRun = true;
+    serverWideCustomAnalyzersUrl = appUrl.forServerWideCustomAnalyzers();
+    canNavigateToServerWideCustomAnalyzers: KnockoutComputed<boolean>;
     
     constructor() {
         super();
         
         aceEditorBindingHandler.install();
         this.bindToCurrentInstance("confirmRemoveAnalyzer", "editAnalyzer");
+
+        this.canNavigateToServerWideCustomAnalyzers = accessManager.default.clusterAdminOrClusterNode;
     }
     
     activate(args: any) {
         super.activate(args);
         
-        return this.loadAnalyzers();
+        return $.when<any>(this.loadAnalyzers(), this.loadServerWideAnalyzers())
+            .done(() => {
+                this.analyzers().forEach(analyzer => {
+                    if (_.includes(this.serverWideAnalyzers().map(x => x.name), analyzer.name)) {
```

Comment:
can we move this mapping outside for loop?

Reply:
done